tests/e2e/test_api.py

```diff
@@ -42,11 +42,12 @@ def test_invalid_question() -> None:
         "/v1/query", json={"conversation_id": "1234", "query": "test query"}, timeout=20
     )
     print(vars(response))
-    assert response.status_code == requests.codes.unprocessable
-    assert response.json() == {
-        "detail": {
-            "response": "Sorry, I can only answer questions about OpenShift "
-            "and Kubernetes. This does not look like something I "
-            "know how to handle."
+    assert response.status_code == requests.codes.ok
+    response.json().get("response") == str(
```

Comment:
please what's the logic behind using and comparing strings, and not deserialized JSON structures directly?

Reply:
Thank you @tisnik Was only testing the response, I updated . Let me see if @bparees has any suggestions on the response object types. I will fix once we confirm.